docs/semgrep-ci/semgrep-ci-with-app.md

```diff
@@ -0,0 +1,332 @@
+        sh 'pip3 install semgrep'
+        sh 'semgrep ci'
+      }
+    }
+  }
+}
+```
+
+</details>
+
+#### Create a `SEMGREP_APP_TOKEN` and set the environment variable in your CI configuration file
+
+To create a `SEMGREP_APP_TOKEN`:
+1. Sign in to Semgrep App.
+2. Click **Settings > Tokens**.
+3. Click **Create new token**.
+4. Copy the name and value, then click **Update**.
+5. Store the token value into your CI provider. Tokens can also be referred to as `secrets`, `credentials`, or `secure variables`. The steps to do this vary depending on your CI provider.
+6. Add the `SEMGREP_APP_TOKEN` environment variable into your Semgrep CI job. Refer to your CI provider's documentation. You can also see the examples in [Create a CI job](#create-a-ci-job-running-semgrep).
+
+#### Grant permissions for Semgrep App (GitHub repositories only)
+
+Follow these steps for GitHub permissions access:
+
+TODO
+
+#### Run the job
+
+Run the job by committing code if it does not start automatically.
+
+#### Set the environment variables
+
+Refer to the following section to set up environment variables.
+
+## Refining the Semgrep App configuration
+
+### Diff-aware scanning
+
+:::tip
+Diff-aware scanning is automatically enabled for all [CI providers listed in Semgrep App](#in-app-providers-such-as-github-actions-gitlab-cicd-jenkins).
+:::
+
+<DiffAwareScanning />
+
+### Enabling hyperlinks to code
+
+:::tip
+Hyperlinks are automatically enabled for all [CI providers listed in Semgrep App](#in-app-providers-such-as-github-actions-gitlab-cicd-jenkins).
+:::
+
+Hyperlinks enable you to view the code that generated the finding from within your repository.
+
+![Screenshot of findings page snippet with no hyperlinks](/img/findings-no-hyperlinks.png "Screenshot of findings page snippet with no hyperlinks")
+*Figure 2.* Partial screenshot of findings page with no hyperlinks.
+
+![Screenshot of findings page snippet with hyperlinks](/img/findings-with-hyperlinks.png "Screenshot of findings page snippet with hyperlinks")
+*Figure 3.* Partial screenshot of findings page with hyperlinks.
+
+To **enable hyperlinks**, the following environment variables must be added into your CI configuration file:
+
+TODO make better examples
+```
+SEMGREP_REPO_NAME="foo/bar"
+SEMGREP_REPO_URL="https://github.com/foo/bar"
+SEMGREP_BRANCH="feature/add-new-bugs"
+SEMGREP_JOB_URL="https://ci-server.com/jobs/1234"
+SEMGREP_COMMIT="a52bc1ef"
+SEMGREP_PR_ID="44"
+```
+
+### Receiving PR or MR comments
+
+:::note
+* Semgrep App does not support PR comments within BitBucket repositories.
+:::
+
+To receive PR or MR comments in your repository, follow the steps to enable hyperlinks. Verify that comments are sent by adding rules to your Rule Board's **Comment** column that can match code to generate a finding.
+
+### Setting a timeout
+To **set a timeout** for the Semgrep job, set the following: TODO
+
+```
+SEMGREP_TIMEOUT
+```
+
+### Customizing rules through the Rule Board
```

Comment:
Link to the rule board page somewhere in this section?

Reply:
Modularized the intro to the rule board from "Getting started with Semgrep App", which has the link.